landscape/lib/sysstats.py

```diff
@@ -110,11 +112,20 @@ class ThermalZone:
     temperature_value = None
     temperature_unit = None
 
-    def __init__(self, base_path, name):
-        self.name = name
-        self.path = os.path.join(base_path, name)
-        temperature_path = os.path.join(self.path, "temp")
-        if os.path.isfile(temperature_path):
+    def __init__(self, temperature_path):
+        self.path = os.path.dirname(temperature_path)
+        self.name = os.path.basename(self.path)
+        if os.path.basename(temperature_path) == "temperature":
+            for line in open(temperature_path):
```

Comment:
Please use `with open` when opening the file. I realize this code was already like this – this is an opportunity to improve it.

Reply:
No problem, fixed!